### PR TITLE
txpool: refactor to use wrapped map for txlist

### PR DESCRIPTION
**Pull request by yoomee1313**
## Proposed changes

- txpool txlist uses `txLookup` instead of `map[common.Hash]*types.Transaction`.
- With this change, it uses the `txLookup` methods to interact with the data inside and those methods are protected by its lock.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: blockchain/tx_pool_test.go [hunk 1]
```diff
@@ -708,7 +708,7 @@ func TestTransactionDropping(t *testing.T) {
 	pool, key := setupTxPool()
 	defer pool.Stop()
 
-	account, _ := deriveSender(transaction(0, 0, key))
+	account := crypto.PubkeyToAddress(key.PublicKey)
```
Comment (kjeom) [inline blockchain/tx_pool_test.go:711]: Does it support klaytn key type?

**Review by blukat29 — APPROVED** [review]
LGTM; It's a syntactic sugar for the lock-protected data structure.

File: blockchain/tx_pool_test.go (hunk 1 above)
Comment (yoomee1313) [reply blockchain/tx_pool_test.go:711]: The `key` type is ecdsa.privateKey, so it is irrelevant with the klaytn key(account) type. That is, only ecdsa.privateKey type is used in test.

Comment (kjeom) [reply blockchain/tx_pool_test.go:711]: Make sense. Thanks for hard working~!

**Review by kjeom — APPROVED** [review]

**Pull request merged**